common/components/common/event_projects/JoinConferenceButton.jsx

```diff
@@ -0,0 +1,45 @@
+// @flow
+
+import React from "react";
+import Button from "react-bootstrap/Button";
+import { Glyph, GlyphSizes, GlyphStyles } from "../../utils/glyphs.js";
+import type { Dictionary } from "../../types/Generics.jsx";
+
+type Props = {|
+  buttonConfig: Dictionary<any>,
+  participant_count: number,
+  className: ?string,
+  variant: string,
+|};
+
+/**
+ * Button for presenting conference links with participant count
+ */
+
+class JoinConferenceButton extends React.PureComponent<Props> {
+  constructor(props: Props): void {
+    super(props);
+  }
+
+  render(): React$Node {
+    return (
+      <React.Fragment>
+        <Button
+          variant={this.props.variant}
```

Comment:
One thing I didn't address was where in the component chain we want to have the safety net for variant -- the `{variant: this.props.variant || default }` -- I think that's done higher up than here right now but for code cleanliness, would it be worth bringing that statement down to this component once and not doing it higher up repeatedly?

Reply:
That scenario should be handled by the buttonConfig prop, which overrides the variant if called for.